fixes #448 updates for new pantheon-quickstart repo and shell scripts (#585)
PegaSysEng/pantheon: docs/Getting-Started/Private-Network-Quickstart.md

# Private Network Quickstart tutorial

!!! important
    In v0.9, the Private Network Quickstart moved to a separate repository, and the previous version was removed from
    the Pantheon repository.

> This tutorial describes how to use Pantheon to run a private network of Pantheon nodes in a Docker container.

Reviewer: ```suggestion
This tutorial describes how to use Pantheon 0.8.5 Docker image to run a private network with multiple Pantheon nodes orchestrated by Docker Compose.
```

!!! note
    To run the Private Network Quickstart, install the code by cloning the `pantheon-quickstart` repo as described below.
    

>     If you have installed Pantheon from the [packaged binaries](../Installation/Install-Binaries.md) or are running the

Author: To be clear, I'm proposing to remove lines 15-16 because the statement is unnecessary and might direct the reader away from the tutorial.

>     If you have installed Pantheon from the [packaged binaries](../Installation/Install-Binaries.md) or are running the

Reviewer 2: Done

    [Docker image](Run-Docker-Image.md), you can proceed with [Starting Pantheon](Starting-Pantheon.md).

## Prerequisites

To run this tutorial, you must have the following installed:

- For MacOS and Linux: [Docker and Docker-compose](https://docs.docker.com/compose/install/) 

- For Windows: [Docker for Windows](https://docs.docker.com/docker-for-windows/install/)

    !!!important

>         Docker for Windows requires Windows 10 Pro, Enterprise, or Education.

Author: Will add "....and a physical Windows computer (not a local or cloud VM)."

        The Private Network Quickstart does not support Docker Toolbox.

- [Git command line](https://git-scm.com/)

- [Curl command line](https://curl.haxx.se/download.html) 

- A web browser that supports [Metamask](https://metamask.io/) (currently Chrome, Firefox, Opera, and Brave), and has 
the MetaMask plug-in installed. This tutorial uses screenshots from Brave.

## Clone Pantheon Source Code

Clone the repository from the `pantheon-quickstart` repo:

```bash
$ git clone --recursive https://github.com/PegaSysEng/pantheon-quickstart.git
```

## Start Services and Network
 
This tutorial uses [Docker Compose](https://docs.docker.com/compose/) to simplify assembling images and 
running in a private network. To build the docker images and run the containers, go to the directory in
which you cloned the `pantheon-quickstart` repo and run:

```bash tab="Linux/MacOS"
# Run the containers:

./run.sh
```

```bash tab="Windows"
// Run the containers:

> .\run.sh

Author: Yes. I verified that in Windows-based command terminals, the scripts run as described above. Git bash is actually a Linux shell; Windows readers who use that will understand it takes the Linux syntax. I just need to add a note that Windows cmd terminal needs to Run as Administrator.

The block explorer displays a summary of the private network:

![Block Explorer](ExplorerSummary.png)

Notice that the explorer indicates 6 peers: the 4 regular nodes, the mining node and the bootnode.

Click the block number to the right of **Best Block** to display the block details:

![Block Details](ExplorerBlockDetails.png)

You can explore blocks by clicking on the blocks under **Bk** on the left-hand side.

You can search for a specific block, transaction hash, or address by clicking the magnifying glass in the top left-hand corner.

![Explorer Search](ExplorerSearch.png)

## Run JSON-RPC Requests 

You can run RPC requests on `rpcnode`, the node exposed to the host in order to listen for requests. This tutorial uses 
[cURL](https://curl.haxx.se/download.html) to make JSON-RPC requests.

!!! tip "Windows"
    We recommend using [Postman](https://www.getpostman.com/) or a similar client to make RPC requests on Windows.
    Using cURL via Command Prompt or Windows PowerShell might not work.

For the RPC URL, this tutorial uses the placeholder `http://localhost:http-rpc-port`. When you run the tutorial, 

> replace `http-rpc-port` with the JSON-RPC HTTP service endpoint provided when you list the endpoints. (For example, 

Reviewer: The place holder is not only port  but the whole URI string.
```suggestion
replace this placeholder with the JSON-RPC HTTP service endpoint provided when you list the endpoints. (For example,
```

```
Here we simply query the version of the Pantheon node, which confirms the node is running.

> 

Reviewer: I liked the idea to tell that if it doesn't work here, no need to go further. How could we tell that differently ?

Author: How about this: "This shows that you can connect to the nodes via RPC. The rest of this section demonstrates more interesting requests. You can walk through them or skip ahead to "Creating a Transaction Using MetaMask."

```

!!!info
    0x79f905c6fd34e80000 = 2250000000000000000000 Wei (2250 Ether).

>     Use a [unit converter](https://etherconverter.online/) to translate values from Wei to Ether.

Reviewer: "you can" seemed appropriate as there's many other ways to do that including manual conversion. Perhaps use a tip there.
```suggestion
!!!tip
    Use a [unit converter](https://etherconverter.online/) to easily translate values from Wei to Ether.
```

Author: Yes--I think it's okay to use "you can" as this is a helpful option.

npm install -g truffle
```

> !!! note

Reviewer: with or without space between `!!!` and `note` it works the same. We could fix all blocks to be consistent in one way or the other, my favourite would be no space.

Author: Agreed. Added this to our Styleguide.

  3 passing (37s)
```

Continue with the Truffle tutorial steps in the [Creating a user interface to interact with the smart contract](https://truffleframework.com/tutorials/pet-shop#creating-a-user-interface-to-interact-with-the-smart-contract) section.

We've already connected the private network to MetaMask, so you can skip the [Installing and configuring MetaMask](https://truffleframework.com/tutorials/pet-shop#installing-and-configuring-metamask) section.

Continue with the regular tutorial steps from the [Installing and configuring lite-server](https://truffleframework.com/tutorials/pet-shop#installing-and-configuring-lite-server)
section and finish the tutorial.

When you adopt pets in the browser and approve the transaction in MetaMask, you'll be able to see the transactions in the block explorer.

## Stop / Restart Private Network without Removing Containers 

To shut down the private network without deleting the containers:

```bash tab="Linux/MacOS"
# Stop the containers:

./stop.sh
```

```bash tab="Windows"
// Stop the containers:

.\stop.sh
```

> !!! note

Reviewer: No, it will only stop the containers related to the services indicated in the docker-compose.yml file
By the way, we talk about services in docker-compose context and containers in docker context.

    This command will also stop other running Docker containers unrelated to the Pantheon quickstart.

To restart the private network:

```bash tab="Linux/MacOS"
# Start the containers:

./start.sh
```

```bash tab="Windows"
// Start the containers:

.\start.sh
```

## Stop Private Network and Remove Containers

> To shut down the private network and delete all containers:

Reviewer: ```suggestion
To shut down the private network and delete all containers and images created during the quickstart:
```

```bash tab="Linux/MacOS"

> # Stop and delete the containers:

Reviewer: ```suggestion
# Stop services and delete related containers and images:
```

./remove.sh
```

```bash tab="Windows"

> // Stop and delete the containers:

Reviewer: ```suggestion
// Stop services and delete related containers and images:
```

.\remove.sh
```

> !!! info "Windows"

Reviewer: It should be. I can add this step in the remove script.
I created the related issue : https://github.com/PegaSysEng/pantheon-quickstart/issues/10
